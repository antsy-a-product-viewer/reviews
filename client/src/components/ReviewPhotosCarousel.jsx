import React from 'react';
import styles from './css/reviewPhotosCarouselStyles.css';

const ReviewPhotosCarousel = (props) => {
  const minIndex = props.page * 5;
  
  const maxIndex = (props.page + 1) * 5;

  const getIndex = (event) => {
    props.openModal(event.target.id);
  };

  if (props.reviewImages.length > 0) {
    return (
      <div style={{height: 120}}>
        {props.reviewImages.slice(minIndex, maxIndex).map((image, index) => {
          return (
            <img className="photoCarouselImg" id={image.review_id} key={image.review_id} onClick={getIndex} style={styles.reviewImage} src={image.review_img}></img>
          );
        })}
      </div>
    );
  } else {
    return null;
  }
};

export default ReviewPhotosCarousel;
