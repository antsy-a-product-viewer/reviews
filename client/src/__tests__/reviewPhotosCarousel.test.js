import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme';
import ReviewPhotosCarousel from '../components/ReviewPhotosCarousel.jsx';

configure({ adapter: new Adapter() });

it('should render correctly with no props', () => {
  const reviewImages = [
    {
      'review_id': 119,
      'img_url': 'https://s3-us-west-1.amazonaws.com/antsyreviewimages/00119.jpg'
    },
    {
      'review_id': 35,
      'img_url': 'https://s3-us-west-1.amazonaws.com/antsyreviewimages/00035.jpg'
    },
    {
      'review_id': 48,
      'img_url': 'https://s3-us-west-1.amazonaws.com/antsyreviewimages/00048.jpg'
    },
    {
      'review_id': 141,
      'img_url': 'https://s3-us-west-1.amazonaws.com/antsyreviewimages/00141.jpg'
    },
    {
      'review_id': 164,
      'img_url': 'https://s3-us-west-1.amazonaws.com/antsyreviewimages/00164.jpg'
    }
  ];
  const props = {
    reviewImages: reviewImages
  };
  const reviewPhotosCarousel = shallow(<ReviewPhotosCarousel reviewImages={props.reviewImages}/>);
  
  expect(reviewPhotosCarousel).toMatchSnapshot();
});