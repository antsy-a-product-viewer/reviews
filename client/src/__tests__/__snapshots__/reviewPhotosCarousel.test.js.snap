// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
<div>
  <img
    key="119"
    style={
      Object {
        "clear": "both",
        "margin": 10,
        "maxHeight": 100,
        "maxWidth": 100,
      }
    }
  />
  <img
    key="35"
    style={
      Object {
        "clear": "both",
        "margin": 10,
        "maxHeight": 100,
        "maxWidth": 100,
      }
    }
  />
  <img
    key="48"
    style={
      Object {
        "clear": "both",
        "margin": 10,
        "maxHeight": 100,
        "maxWidth": 100,
      }
    }
  />
  <img
    key="141"
    style={
      Object {
        "clear": "both",
        "margin": 10,
        "maxHeight": 100,
        "maxWidth": 100,
      }
    }
  />
  <img
    key="164"
    style={
      Object {
        "clear": "both",
        "margin": 10,
        "maxHeight": 100,
        "maxWidth": 100,
      }
    }
  />
</div>
`;
